test/integrations/destinations/klaviyo/processor/ecomTestData.ts

```diff
@@ -0,0 +1,344 @@
+import { overrideDestination, transformResultBuilder } from '../../../testUtils';
+
+const destination = {
+  Config: {
+    publicApiKey: 'dummyPublicApiKey',
+    privateApiKey: 'dummyPrivateApiKey',
+  },
+};
+
+const commonTraits = {
+  id: 'user@1',
+  age: '22',
+  name: 'Test',
+  email: 'test@rudderstack.com',
+  phone: '9112340375',
+  anonymousId: '9c6bd77ea9da3e68',
+  description: 'Sample description',
+};
+
+const eventsEndpoint = 'https://a.klaviyo.com/api/events';
+
+const commonOutputHeaders = {
+  Authorization: 'Klaviyo-API-Key dummyPrivateApiKey',
+  'Content-Type': 'application/json',
+  Accept: 'application/json',
+  revision: '2023-02-22',
+};
+
+export const ecomTestData = [
+  {
+    id: 'klaviyo-ecom-test-1',
+    name: 'klaviyo',
+    description: 'Track call with Ecom events (Viewed Product)',
+    scenario: 'Business',
+    successCriteria:
+      'Response should contain only event payload and status code should be 200, for the event payload should contain contextual traits and properties in the payload the event name should be Viewed Product and the properties should be mapped to the Klaviyo event properties',
+    feature: 'processor',
+    module: 'destination',
+    version: 'v0',
+    input: {
+      request: {
+        body: [
+          {
+            destination,
+            message: {
+              type: 'track',
+              event: 'product viewed',
+              sentAt: '2021-01-25T16:12:02.048Z',
+              userId: 'sajal12',
+              channel: 'mobile',
+              context: {
+                traits: commonTraits,
+              },
+              properties: {
+                name: 'test product',
+                product_id: '1114',
+                sku: 'WINNIePuh12',
+                image_url: 'http://www.example.com/path/to/product/image.png',
+                url: 'http://www.example.com/path/to/product',
+                brand: 'Not for Kids',
+                price: 9.9,
+                categories: ['Fiction', 'Children'],
+              },
+              anonymousId: '9c6bd77ea9da3e68',
+              originalTimestamp: '2021-01-25T15:32:56.409Z',
+            },
+          },
+        ],
+      },
+    },
+    output: {
+      response: {
+        status: 200,
+        body: [
+          {
+            output: transformResultBuilder({
```

Comment:
how is this `transformResultBuilder` utility helping out ? By restricting only to required fields in the util function, is this not hiding any wrong implementation by the destination ?

Reply:
If you check the implementation of this utility it generates the response payload with with only the required fields the rest of the boiler plate fields are auto generated. If there is any wrong implementation that would be caught by the diff. One has to make sure from a business perspective the required fields are correct or now
https://github.com/rudderlabs/rudder-transformer/blob/7ae320c275bb4813ceaa763df0798075eee18cfb/test/integrations/testUtils.ts

Reply:
We need to update this boilerplate whenever there are any updates to API's like the `config field getting introduced in proxy` & each API should have it's associated resultBuilder method